sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -363,7 +449,7 @@ private int calculateTotalSerializedLength(AtomicInteger currentTotalSerializedL
         FluxSink<CosmosItemOperation> groupSink) {
 
         if (operations.size() == 0) {
-            logger.debug("Empty operations list, Context: {}", this.operationContextText);
+            logger.trace("Empty operations list, Context: {}", this.operationContextText);
```

Comment:
why changing to trace? how we decided what should be trace vs debug? (sometimes I find it is hard for myself to decide)

Reply:
I used trace on very noisy logs that I assume won't be too valuable for most investigations. I used debug for noisy logs that could be more interesting (either in the main flow, about completion or errors (even when retried handled). Most of the trace logs I left were added when I tried to narrow down where exactly the processing stopped.